cfy_manager/utils/common.py

```diff
@@ -38,16 +40,17 @@ def run(command, retries=0, stdin=b'', ignore_failures=False,
         for arg in command:
             glob_command.append(glob.glob(arg))
         command = glob_command
-    logger.debug('Running: {0}'.format(command))
+    logger.debug('Running: {0}'.format(command if log_command else '<hidden>'))
     proc = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=stdout,
                             stderr=stderr, shell=shell, env=env)
     proc.aggr_stdout, proc.aggr_stderr = proc.communicate(input=stdin)
     if proc.returncode != 0:
-        command_str = ' '.join(command)
+        command_str = ' '.join(command) if log_command else '<hidden>'
         if retries:
             logger.warn('Failed running command: {0}. Retrying. '
                         '({1} left)'.format(command_str, retries))
-            proc = run(command, retries - 1)
+            time.sleep(retry_interval)
+            proc = run(command, retries - 1, retry_interval=retry_interval)
```

Comment:
we need to pass other arguments here as well, especially env (wont work without it) and log_command (commands might get logged even though we wanted them to be hidden)

but i think it's odd that the retry is recrusive, instead of a loop with a break

Reply:
@tehasdf no recursion anymore.